### PR TITLE
fix: ensures metadata is present before checking the content of the grants property

### PR DESCRIPTION
**Pull request by gdixon**
This PR will fix an error on the cart where we were checking the contents of the metadata before it was available/resolved.

--

Fixes: #577

### CODE REVIEW

**Review by mk1r — APPROVED** [review]
nice fix! 👌

File: app/src/store/cart.ts [hunk 1]
```diff
@@ -393,7 +393,8 @@ export default function useCartStore() {
     return cart.value
       .map((item) =>
         (grantRounds.value || []).reduce((inRound: boolean, round) => {
-          return inRound || grantRoundMetadata.value[metadataId(round.metaPtr)].grants?.includes(item.grantId) || false;
+          const metadata = grantRoundMetadata.value[metadataId(round.metaPtr)];
+          return inRound || metadata?.grants?.includes(item.grantId) || false;
```
Comment (mds1) [inline app/src/store/cart.ts:397]: Sanity check—the fix here is the added `?` before accessing the `grants` key right?

**Review by mds1 — APPROVED** [review]

File: app/src/store/cart.ts (hunk 1 above)
Comment (mds1) [reply app/src/store/cart.ts:397]: And this means #577 was not actually related to a user adding "Grants which are both in a round and not in a round to their cart and then refreshes the page"? i.e. it's seems like this just has to do with refreshing on the cart page?

Comment (gdixon) [reply app/src/store/cart.ts:397]: Thats correct, I wasn't exactly sure what was going on when I wrote the issue up 😅

**Pull request merged**